scripts/drag-n-bar.js

```diff
@@ -943,48 +1076,130 @@ H5P.DragNBar.prototype.add = function ($element, clipboardData, options) {
+    if(typeof element.$element.attr('class').split(" ").find(cName => cName.startsWith("h5p-dnb-unique-")) !== 'string') {
+    
+    const uniqueClassFloat = Math.random();
+    
+    // Adding class to element
+    const startStringElement = 'h5p-dnb-unique-';
+    const uniqueClassString = startStringElement + uniqueClassFloat.toString(32);
+    element.$element.addClass(uniqueClassString);
+    
+    // Adding control-box
+    const startStringControlBox = 'h5p-control-box-unique-';
+    const uniqueControlBox = startStringControlBox + uniqueClassFloat.toString(32);
+    self.createMoveableControlBoxOnElement(element.$element, uniqueControlBox);
+
+    // Hiding moveable-control-boxes. This is because if we edit a whole CP (from the menu), we dont want all the boxes to show.
+    // If we are just adding an element, it will get 'focused' after this code is run, so it's ok.
+    self.hideControlBoxes();
+
+    // Moving the control-box to overlap the element.
+    const rectElementBCR = element.$element[0].getBoundingClientRect();
+    const theControlBoxElement = document.getElementsByClassName(uniqueControlBox)[0];
+    const dndToolbarHeight = 38;
+    theControlBoxElement.style.transform = `translate3d(${rectElementBCR.left + window.scrollX}px, ${rectElementBCR.top + window.scrollY + dndToolbarHeight}px, 0px)`;
+    }
+  }
+}
+
+// Cleaning up all control-boxes which are not in use
+H5P.DragNBar.prototype.removeControlBoxesNotInUse = function () {
+
+  //Getting unique ID's from elements on all slides in the CP, which are the same ID's to the corresponding control-boxes
+  let uniqueClassStringList = [];
+  const wrapper = document.getElementsByClassName('h5p-slides-wrapper');
+  for (let i = 0; i < wrapper[0].childNodes.length; i++) {
+    for (let y = 0; y < wrapper[0].childNodes[i].childNodes.length; y++) {
+      if(typeof wrapper[0].childNodes[i].childNodes[y].classList.value
+        .split(" ").find(cName => cName.startsWith("h5p-dnb-unique-")) != 'undefined') {
+        uniqueClassStringList
+          .push(wrapper[0].childNodes[i].childNodes[y].classList.value
+          .split(" ")
+          .find(cName => cName.startsWith("h5p-dnb-unique-"))
+          .split("-").pop());
+      }
+    }
+  }
+  
+  // Removing all control-boxes which are not active anymore
+  const x = document.getElementsByClassName('moveable-control-box');
+  // Iterate through alle the control-box-elements
+  for (let i = x.length - 1; i >= 0; i--) {
+    let found = false;
+    // Looking for a match from the list
+    for (let y = 0; y < uniqueClassStringList.length; y++) {
+      if (uniqueClassStringList[y] == (x[i].classList.value.split(" ").find(cName => cName.startsWith("h5p-control-box-unique-")).split("-").pop())) {
+        found = true;
+        break;
+      }
+    }
+    // If not found, the control-box does not have a corresponding element. Therefore we remove.
+    if(!found) {
+      x[i].remove();
+    }
+  }
```

Comment:
I'm not sure if it is accurate, but can we not do it like this?

```suggestion
  for (const uniqueClass of uniqueClassStringList) {
    const controlBox = document.querySelector(`.moveable-control-box.h5p-control-box-unique-${uniqueClass}`);
    if (!controlBox) {
      controlBox.parentElement.removeChild(controlBox);
    }
  }
```

Reply:
They way you suggest you are finding a control-box that has an existing element. The if-statement will return true in all cases, since uniqueClassStringList is a list of unique-ids of the existing elements in the scene, and there will always be a control-box corresponding to those. I want to remove the control-boxes which do not have a correlating element in the scene. There probably is a better way to do it than I have done. Im all ears :)